### ISSUE TITLE
Weird behaviour of draw_segmentation_masks

### ISSUE THREAD

**Issue by gungui98** [pytorch/vision#4584]
### 🐛 Describe the bug

from torchvision.utils import draw_segmentation_masks
import torch
import cv2
import numpy as np

image = np.ones((255, 255, 3), dtype=np.uint8)*255
mask = np.zeros((255, 255), dtype=np.bool)
mask[40:100, 20: 40] = True 

image_tensor = torch.from_numpy(image)
h, w, c = image_tensor.shape
image_tensor = image_tensor.view(c, h, w)
mask_tensor = torch.from_numpy(mask)

print(image_tensor.shape)
print(mask_tensor.shape)

image_mask = draw_segmentation_masks(image_tensor, mask_tensor, alpha=1, colors = [(255, 0, 0), (0, 255, 0)])
image_mask = image_mask.view(h, w, c).numpy()

# I was expecting a single image with mask overlay but this show a grid of 6 images`
cv2.imwrite("test_draw.jpg", image_mask)

### Versions

Collecting environment information...
PyTorch version: 1.9.1+cu111
Is debug build: False
CUDA used to build PyTorch: 11.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.3 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.9.7 (default, Sep 16 2021, 13:09:58)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.11.0-34-generic-x86_64-with-glibc2.31
Is CUDA available: True
CUDA runtime version: 10.1.243
GPU models and configuration:
GPU 0: NVIDIA GeForce GTX 1060 6GB
GPU 1: NVIDIA GeForce GTX 1070

Nvidia driver version: 470.63.01
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.21.2
[pip3] pytorch-lightning==1.4.9
[pip3] torch==1.9.1+cu111
[pip3] torchaudio==0.9.1
[pip3] torchmetrics==0.5.1
[pip3] torchvision==0.10.1+cu111
[conda] numpy                     1.21.2                   pypi_0    pypi
[conda] pytorch-lightning         1.4.9                    pypi_0    pypi
[conda] torch                     1.9.1+cu111              pypi_0    pypi
[conda] torchaudio                0.9.1                    pypi_0    pypi
[conda] torchmetrics              0.5.1                    pypi_0    pypi
[conda] torchvision               0.10.1+cu111             pypi_0    pypi
